base/tuple.jl

```diff
@@ -265,11 +266,19 @@ function isequal(t1::Tuple, t2::Tuple)
     return true
 end
 
-function ==(t1::Tuple, t2::Tuple)
-    if length(t1) != length(t2)
+==(t1::Tuple, t2::Tuple) = (length(t1) == length(t2)) && _eq(t1, t2, false)
+_eq(t1::Tuple{}, t2::Tuple{}, anymissing) = anymissing ? missing : true
+function _eq(t1::Tuple, t2::Tuple, anymissing)
+    eq = t1[1] == t2[1]
+    if ismissing(eq)
+        return _eq(tail(t1), tail(t2), true)
+    elseif !eq
         return false
+    else
+        return _eq(tail(t1), tail(t2), anymissing)
```

Comment:
I think this can be simplified to

```
if eq === false
    return false
else
    return _eq(tail(t1), tail(t2), anymissing | ismissing(eq))
end
```